### PR TITLE
Rename tpp and xsmm matmul ops

### PR DESCRIPTION
**Pull request by chelini**
Rename `tpp.matmul` to `tpp.gemm` and `xsmm.matmul` to `xsmm.gemm`. This better reflects the naming used
in LIBXSMM. `tpp.gemm` assumes beta = 1.

### CODE REVIEW

**Comment by chelini** [comment]
depends on #498

File: test/Dialect/Tpp/tpp-invalid.mlir [hunk 1]
```diff
@@ -310,14 +309,6 @@ func.func @tpp_relu_invalid_number_of_operands(%arg0: tensor<2x2xf32>) -> tensor
 
 // -----
 
-func.func @tpp_matmul_invalid_number_of_operands(%arg0: tensor<2x2xf32>) -> tensor<2x2xf32> {
```
Comment (chelini) [inline test/Dialect/Tpp/tpp-invalid.mlir:313]: this test was duplicated.

Comment (adam-smnk) [reply test/Dialect/Tpp/tpp-invalid.mlir:313]: Isn't the other one on memref?

Comment (chelini) [reply test/Dialect/Tpp/tpp-invalid.mlir:313]: Good catch, sorry. Added back

**Comment by rengolin** [comment]
> For optimization purposes we will need a new `tpp.matmul` where the beta equals 0, thus rename the current `tpp.matmul` to `tpp.gemm`.

We don't need `tpp.matmul`, we'll have `tpp.group { tpp.zero + tpp.gemm }` which is the same thing without the need of a new op.

> `tpp.gemm` now assumes beta = 1.

It shouldn't.

**Comment by rengolin** [comment]
> > `tpp.gemm` now assumes beta = 1.
> 
> It shouldn't.

Correction: it should. This is the safest option, and would generate less efficient code if it actually executed `tpp.zero` first.

Update to beta = 0 if there's a `tpp.zero` before can be done as an optimization.

Temporarily we could have a flag (which would be a replacement from `tpp.zero+tpp.gemm -> tpp.gemm { beta = 0 }` anyway. 

But this should *NOT* be the long term solution, as we'll have `tpp.group` for the very reason to not need to create too many ops and attributes.

**Pull request merged**